docs/adr/0002-shopping-lists-lineitem-and-textlineitem-update-actions.md

```diff
@@ -0,0 +1,336 @@
+                the order will be <b>SKU-1</b> and <b>SKU-3</b>.</p>
+            <p>The <b>challenge</b> in here is, those actions can not be added in one request because we don't know the
+                line item id of the new line item
+                with <b>SKU-3</b>, so we need to find another way to create a new line item with the right order.</p>
+            <p>Also another <b>challenge</b> in here is about the line item with <b>SKU-1</b>, as the order and data is
+                exactly same, we need to find a better way to avoid creating unnecessary actions.
+            </p>
+        </td>
+    </tr>
+    <tr>
+        <th colspan="2">Proposed solution</th>
+    </tr>
+    <tr>
+        <td colspan="2">
+            <p>
+                The solution idea about the new line item and changed order is still same with the case-1, but do we
+                need to remove and add line item with <b>SKU-1</b>? Which is not needed and we could start the removing
+                and adding from the changed order.
+            </p>
+            <pre lang="json">
+{
+  "version": 1,
+  "actions": [
+    {
+      "action": "removeLineItem",
+      "lineItemId": "24de3821-e27d-4ddb-bd0b-ecc99365285f"
+    },
+    {
+      "action": "addLineItem",
+      "sku": "SKU-3",
+      "quantity": 3,
+      "addedAt": "2020-11-05T10:00:10.101Z",
+      "custom": {
+        "type": {
+          "key": "custom-type-for-shoppinglists"
+        },
+        "fields": {
+          "textField": "text-3
+        }
+      }
+    }
+  ]
+}
+</pre>
+        </td>
+    </tr>
+</table>
+
+### How addedAt will be compared?
+
+In commercetools shopping lists API, there is no [update action](https://docs.commercetools.com/api/projects/shoppingLists#update-actions) 
+to change the `addedAt` field of the `LineItem` and `TextLineItem`, also in API it has a default value described as `Defaults to the current date and time.`, 
+when it's not set in the draft, so how to compare and update this field?
+
+**Proposed solution:**
+
+A sync option for excluding `addedAt` (based on the draft) comparisons, which defaults to `true`. The advantage of this, customers could decide themself sync `addedAt`. 
+The drawback is when the option set to `false` and a difference is found by our utils, to ensure the order we need to remove and add all line items.  
+
+> Note: `addedAt` data might be important if the customer has a front end that sorts the line items with the date.
+
+## Decision (not agreed yet with the team)
+
+<!-- The change that we're proposing or have agreed to implement. -->
+
+- Product variant will be matched by its SKU, if the SKU not set for a LineItemDraft, the draft will not be synced and an error callback will be triggered.
+> Note: In commercetools API, the product variant to be selected in the LineItemDraft can be specified either by its product ID plus variant ID or by its SKU.
```

Comment:
I see there's no SKU 😕 ? https://docs.commercetools.com/api/projects/shoppingLists#add-lineitem

Reply:
I had a look again and SKU is only mentioned when you create a shopping list, not when you update... ❓  😕

Reply:
Drafts are always for creation, update case is not an issue, because you will know the `lineItemId` coming from the target resource. Maybe the confusing part is related to sync library because we expect draft even it exists in the target, in this case, the library user needs to provide SKU with extending variants of the line items, otherwise, validation will trigger callback. I will add the link here: https://docs.commercetools.com/api/projects/shoppingLists#lineitemdraft-product-variant-selection 

https://github.com/commercetools/commercetools-sync-java/pull/614/commits/3fba6c514c23d684139f548e41ecc5c2091b3650

Reply:
Now looks better: https://github.com/commercetools/commercetools-sync-java/pull/614/commits/7c9fc7a76abedc56ed6043bd602ac932c0fb7400